#!/usr/bin/env python
PACKAGE="vimjay"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("capture_single", bool_t, 1, "capture single image", False)
gen.add("update_rate", double_t, 2, "updates per second", 4.0, 0.001, 30.0)
gen.add("pause", bool_t, 8, "pause", False)
gen.add("next", bool_t, 8, "move to next frame if paused", False)
gen.add("prev", bool_t, 8, "move to previous frame if paused", False)
gen.add("remove", bool_t, 8, "remove current frame if paused", False)
gen.add("index", int_t, 4, "current index", 0, 0, 512)
gen.add("start_index", int_t, 8, "start index", 0, 0, 512)
gen.add("start_index_to_end", bool_t, 8, "move start index to end of deque", False)
gen.add("use_live_frame", bool_t, 8, "use live frame as final frame of animation", True)
gen.add("capture_continuous", bool_t, 8, "save all received images", False)
gen.add("restrict_size", bool_t, 8, "limit deque size", False)
gen.add("max_size", int_t, 8, "limit deque size", 10, 1, 512)

exit(gen.generate(PACKAGE, "image_deque", "ImageDeque"))
